### ISSUE TITLE
Attempt at running force recovery using 2.x snap on 1.10 instance results in crash

### ISSUE THREAD

**Issue by Korablev77** [tarantool/tarantool#5612]
```
Tarantool 1.10.7-48-gb3cb182e5
type 'help' for interactive help
tarantool> box.cfg{force_recovery=true}
2020-12-08 13:03:37.427 [15583] main/101/interactive C> Tarantool 1.10.7-48-gb3cb182e5
2020-12-08 13:03:37.427 [15583] main/101/interactive C> log level 5
2020-12-08 13:03:37.428 [15583] main/101/interactive I> mapping 268435456 bytes for memtx tuple arena...
2020-12-08 13:03:37.428 [15583] main/101/interactive I> mapping 134217728 bytes for vinyl tuple arena...
2020-12-08 13:03:37.443 [15583] main/101/interactive I> instance uuid 88f97fdb-10b0-4aad-bbf4-26927bec13cc
2020-12-08 13:03:37.444 [15583] main/101/interactive I> instance vclock {1: 5}
2020-12-08 13:03:37.445 [15583] main/101/interactive I> recovery start
2020-12-08 13:03:37.445 [15583] main/101/interactive I> recovering from `./00000000000000000000.snap'
2020-12-08 13:03:37.481 [15583] main/101/interactive I> cluster uuid 2919f050-b40c-4046-bf0d-3aab9f2900fb
2020-12-08 13:03:37.481 [15583] main/101/interactive memtx_engine.c:233 E> can't apply row: 
2020-12-08 13:03:37.481 [15583] main/101/interactive alter.cc:2380 E> ER_CANT_CREATE_COLLATION: Failed to initialize collation: unknown collation type.
2020-12-08 13:03:37.481 [15583] main/101/interactive memtx_engine.c:233 E> can't apply row: 
2020-12-08 13:03:37.481 [15583] main/101/interactive alter.cc:2380 E> ER_CANT_CREATE_COLLATION: Failed to initialize collation: unknown collation type.
...
2020-12-08 13:03:37.491 [15583] main/101/interactive memtx_engine.c:233 E> can't apply row: 
2020-12-08 13:03:37.491 [15583] main/101/interactive alter.cc:2272 E> ER_FUNCTION_LANGUAGE: Unsupported language 'SQL_BUILTIN' specified for function '_sql_stat_init'
2020-12-08 13:03:37.491 [15583] main/101/interactive memtx_engine.c:233 E> can't apply row: 
2020-12-08 13:03:37.491 [15583] main/101/interactive alter.cc:2272 E> ER_FUNCTION_LANGUAGE: Unsupported language 'SQL_BUILTIN' specified for function 'GREATEST'
2020-12-08 13:03:37.491 [15583] main/101/interactive memtx_engine.c:233 E> can't apply row: 
2020-12-08 13:03:37.491 [15583] main/101/interactive alter.cc:2272 E> ER_FUNCTION_LANGUAGE: Unsupported language 'SQL_BUILTIN' specified for function 'LEAST'
tarantool: tarantool/src/box/schema.cc:612: const char* schema_find_name(schema_object_type, uint32_t): Assertion `false' failed.
Aborted (core dumped)
```

**Comment by ImeevMA** [comment]
At the moment, the error has not disappeared, but changed.

I run there commands on current master (`Tarantool 2.10.0-beta2-157-gc86145b33`):
```
box.cfg{}
box.schema.space.create('a')
box.snapshot()
```

Then moved created snapshot to current 1.10 (`Tarantool 1.10.12-45-gdfb873cf5`) and run this command:
```
box.cfg{}
```

The result is:
```
Tarantool 1.10.12-45-gdfb873cf5
type 'help' for interactive help
tarantool> box.cfg{}
2022-03-11 13:10:55.583 [217793] main/101/interactive C> Tarantool 1.10.12-45-gdfb873cf5
2022-03-11 13:10:55.583 [217793] main/101/interactive C> log level 5
2022-03-11 13:10:55.583 [217793] main/101/interactive I> mapping 268435456 bytes for memtx tuple arena...
2022-03-11 13:10:55.583 [217793] main/101/interactive I> Actual slab_alloc_factor calculated on the basis of desired slab_alloc_factor = 1.044274
2022-03-11 13:10:55.583 [217793] main/101/interactive I> mapping 134217728 bytes for vinyl tuple arena...
2022-03-11 13:10:55.589 [217793] main/101/interactive I> instance uuid dd55ebac-087c-4aed-9930-0070bb96e7e7
2022-03-11 13:10:55.589 [217793] main/101/interactive I> instance vclock {1: 2}
2022-03-11 13:10:55.590 [217793] main/101/interactive I> recovery start
2022-03-11 13:10:55.590 [217793] main/101/interactive I> recovering from `./00000000000000000002.snap'
2022-03-11 13:10:55.608 [217793] main/101/interactive I> cluster uuid a3fd6880-ae18-4091-bc77-a863b41ae388
2022-03-11 13:10:55.608 [217793] main/101/interactive alter.cc:2379 E> ER_CANT_CREATE_COLLATION: Failed to initialize collation: unknown collation type.
2022-03-11 13:10:55.608 [217793] main/101/interactive F> can't initialize storage: Failed to initialize collation: unknown collation type.
```

**Comment by kyukhin** [comment]
Fixed by [0fb704f](https://github.com/tarantool/tarantool/commit/0fb704fb0d5f6d3f30b79e37f83552d2e79dd642).
